### ISSUE TITLE
How to catch the "Uncaught Error: Timed out waiting 10000ms for payment" exception ?

### ISSUE THREAD

**Issue by aledelgo** [paypal/paypal-checkout#460]
I'm using the `paypal.Button.render` function to render a payment button.
Some time my app server may take more time than 10 seconds to generate a paypal payment id and i get this error:

```
Uncaught Error: Timed out waiting 10000ms for payment
    at ParentComponent.<anonymous> (https://www.paypalobjects.com/api/checkout.js:8769:152)
    at Object.value (…

```

I just would like to catch that kind of exception so i could tell the user to try again.

(yes i know i could retrive the payment id before rendering the button but in my case this is not an option as for example i shouldn't process such as that heavy request if the user choose other payment methods...)

thanks much

**Comment by bluepnume** [comment]
Out of interest, what is taking more than 10 seconds? The paypal payments api should usually return a lot faster than that.

**Comment by aledelgo** [comment]
In my project the payment should be capture only in certain conditionsand refused in all the other... conditions are checked by the appserver in high concurrency where multiple async processes (and users) may lock records on DB to avoid collision and/or perform other webrequests... this add time to paypal request that anyway, in sandbox, from my local PC take about 3 seconds (when WebProfileId is cached)...

**Comment by bluepnume** [comment]
Would it be possible for you to do all of the above in parallel somehow? i.e. send the request to set up the paypal payment, and do everything else in a different request which does not block the PayPal transaction from continuing?

The problem is, after 10 seconds of a user seeing the PayPal spinner, a high percentage of them start to drop off, and close the window. So waiting longer than 10 seconds ends up being highly detrimental to conversion.

**Comment by aledelgo** [comment]
Yes, it is how i  actually solved the problem but it doesn't change the user experience too much... Actually when page load the request start and when ready paypal button is shown.... But this mean make a lot of unuseful heavy requests also when user want to pay with a different method...

Also I've tried to make the request when user hit my custom button but then i cannot start the paypal button click from code...

**Comment by bluepnume** [comment]
Would it be possible for you to do something like:

```javascript
paypal.Button.render({
    payment() {
        // Call server to call PayPal api
        return paypal.request.post('/create-paypal-payment').then(function(data) {

            // Make a fire-and-forget call to server to update database, etc.
            paypal.request.post('/update-database', { paymentID: data.paymentID });

            // Immediately return the payment id so we can launch the paypal flow in the popup window
            return data.paymentID;
        });
    }
});
```

Where `/create-paypal-payment` is responsible for *only* calling the PayPal api and returning the payment id, and `/update-database` is responsible for updating your own database, etc?

**Comment by bluepnume** [comment]
Closing as duplicate of https://github.com/paypal/paypal-checkout/issues/477
